### ISSUE TITLE
Unable to run Merlin with GPU

### ISSUE THREAD

**Issue by martedva** [CSTR-Edinburgh/merlin#315]
Hello,

So after successfully having run merlin on CPU, I would like to switch to my GPU, hereby allowing for faster training.

The problem is that _merlin_ is able to detect the GPU (Running on GPU id=0 ...), however, it also says, that it is using **cuDNN version 7101 on context None**, and maps None to the device cuda0.

![image](https://user-images.githubusercontent.com/6831315/37572109-8c61975a-2b06-11e8-9fe5-b31a138c9de3.png)

I have attached the above picture of the problem, which I encounter. At also tells me that it has created my demo voice, however, it has in fact not.

I have gone through all issues in here about GPU problems, such as Theano flags, cuDNN installation, libgpuarray, and such. I have run out of ideas and links on Google. 

Does anyone have an idea?

Thanks in advance.

**Comment by seblemaguer** [comment]
The error is not about the gpu but the config file. Did you run the setup properly?

**Comment by martedva** [comment]
@seblemaguer I will check this as soon as I get home, and get back to you!

**Comment by martedva** [comment]
Hi, @seblemaguer 
So I ran the steps in slt_arctic_full by running each script individually. It now actually worked and seemed to run on the GPU, however, it still gives me the same line:

![image](https://user-images.githubusercontent.com/6831315/37593888-f3abef10-2b73-11e8-9cc6-56a34fba527e.png)

I looked at my config-file, and everything seems to be looking right. I haven't changed a thing, but this should not be necessary, as I am merely running the slt_arctic_full. Nonetheless I attached my config file too.

**global_settings.cfg**
![image](https://user-images.githubusercontent.com/6831315/37593999-461f9346-2b74-11e8-8849-f5781ceada65.png)

**acoustic_slt_arctic_full.conf**
`[DEFAULT]

Merlin: /home/tts/merlin

TOPLEVEL: /home/tts/merlin/egs/slt_arctic/s1

[Paths]

# where to place work files
work: %(TOPLEVEL)s/experiments/slt_arctic_full/acoustic_model

# where to find the data
data: %(work)s/data

# where to find intermediate directories
inter_data: %(work)s/inter_module

# list of file basenames, training and validation in a single list
file_id_list: %(data)s/file_id_list_full.scp
test_id_list: %(data)s/test_id_list.scp

in_mgc_dir: %(data)s/mgc
in_bap_dir: %(data)s/bap
in_lf0_dir: %(data)s/lf0

# where to save log
log_path: %(work)s/log

# where to save plots
plot: %(work)s/plots

# logging
log_config_file: %(TOPLEVEL)s/conf/logging_config.conf
log_file: %(work)s/log/mylogfilename.log

# where are my tools
sptk:  %(Merlin)s/tools/bin/SPTK-3.9
straight: %(Merlin)s/tools/bin/straight
world: %(Merlin)s/tools/bin/WORLD
magphase: %(Merlin)s/tools/magphase/src

[Labels]
enforce_silence: False
silence_pattern: ['*-sil+*']
# options: state_align or phone_align
label_type: state_align
label_align: %(TOPLEVEL)s/experiments/slt_arctic_full/acoustic_model/data/label_state_align
question_file_name: %(Merlin)s/misc/questions/questions-radio_dnn_416.hed

add_frame_features: True

# options: full, coarse_coding, minimal_frame, state_only, frame_only, none
subphone_feats: full

[Outputs]
# dX should be 3 times X
mgc: 60
dmgc: 180
bap: 1
dbap: 3
lf0: 1
dlf0: 3

[Waveform]
test_synth_dir: None
# options: WORLD, STRAIGHT or MAGPHASE
vocoder_type: WORLD
samplerate: 16000
framelength: 1024
# Frequency warping coefficient used to compress the spectral envelope into MGC (or MCEP).
fw_alpha: 0.58
minimum_phase_order: 511
use_cep_ap: True

[Architecture]
switch_to_tensorflow: False
switch_to_keras: False
hidden_layer_size  : [1024, 1024, 1024, 1024, 1024, 1024]
hidden_layer_type  : ['TANH', 'TANH', 'TANH', 'TANH', 'TANH', 'TANH']

model_file_name: feed_forward_6_tanh

#if RNN or sequential training is used, please set sequential_training to True.
sequential_training : False

dropout_rate : 0.0
batch_size   : 256

# options: -1 for exponential decay, 0 for constant learning rate, 1 for linear decay
lr_decay      : -1 
learning_rate : 0.002

# options: sgd, adam, rprop
optimizer : sgd

warmup_epoch    : 10
training_epochs : 25

[Streams]
# which feature to be used in the output
output_features      : ['mgc', 'lf0', 'vuv', 'bap']
gen_wav_features     : ['mgc', 'lf0', 'bap']

[Data]
train_file_number: 1000
valid_file_number: 66
test_file_number: 66
#buffer size of each block of data to
buffer_size: 200000

[Processes]

# Main processes

AcousticModel : True
GenTestList : False

# sub-processes

NORMLAB  : True
MAKECMP  : True
NORMCMP  : True

TRAINDNN : True
DNNGEN   : True

GENWAV   : True
CALMCD   : True`

**Comment by seblemaguer** [comment]
can you use triple quotes to format the source code you are using it is really difficult to read else (here is the doc: https://help.github.com/articles/creating-and-highlighting-code-blocks/)

I am not talking about the global config but about the `conf/test_synth_slt_arctic_full.conf`. The screenshot of your first post indicates that this configuration doesn't exist

**Comment by martedva** [comment]
Hello @seblemaguer,

So I successfully created my slt_arctic_full voice, but I now encounter a new problem regarding the GPU usage. As I said, it now actually works on the GPU, it seems, as it is WAY faster now.

However, I now get the following error:
![image](https://user-images.githubusercontent.com/6831315/37600141-bcc174bc-2b86-11e8-87ae-a3f3e1233915.png)

I have tried to research this problem, and most of the links tell me that I need to update _theano_ to the newest beta version, however, I already have version 1.0.1, which seems to be the newest.

My conf/test_synth_slt_arctic_full.conf file looks as follows:
```
[DEFAULT]

Merlin: /home/tts/merlin

TOPLEVEL: /home/tts/merlin/egs/slt_arctic/s1

[Paths]

# where to place work files
work: %(TOPLEVEL)s/experiments/slt_arctic_full/acoustic_model

# where to find the data
data: %(work)s/data

# where to find intermediate directories
inter_data: %(work)s/inter_module

# list of file basenames, training and validation in a single list
file_id_list: %(data)s/file_id_list_full.scp
test_id_list: %(TOPLEVEL)s/experiments/slt_arctic_full/test_synthesis/test_id_list.scp

in_mgc_dir: %(data)s/mgc
in_bap_dir: %(data)s/bap
in_lf0_dir: %(data)s/lf0

# where to save log
log_path: %(work)s/log

# where to save plots
plot: %(work)s/plots

# logging
log_config_file: %(TOPLEVEL)s/conf/logging_config.conf
log_file: %(work)s/log/mylogfilename.log

# where are my tools
sptk:  %(Merlin)s/tools/bin/SPTK-3.9
straight: %(Merlin)s/tools/bin/straight
world: %(Merlin)s/tools/bin/WORLD
magphase: %(Merlin)s/tools/magphase/src

[Labels]
enforce_silence: True
silence_pattern: ['*-sil+*']
# options: state_align or phone_align
label_type: state_align
label_align: %(TOPLEVEL)s/experiments/slt_arctic_full/test_synthesis/gen-lab
question_file_name: %(Merlin)s/misc/questions/questions-radio_dnn_416.hed

add_frame_features: True

# options: full, coarse_coding, minimal_frame, state_only, frame_only, none
subphone_feats: full

[Outputs]
# dX should be 3 times X
mgc: 60
dmgc: 180
bap: 1
dbap: 3
lf0: 1
dlf0: 3

[Waveform]
test_synth_dir: %(TOPLEVEL)s/experiments/slt_arctic_full/test_synthesis/wav
# options: WORLD, STRAIGHT or MAGPHASE
vocoder_type: WORLD
samplerate: 16000
framelength: 1024
# Frequency warping coefficient used to compress the spectral envelope into MGC (or MCEP).
fw_alpha: 0.58
minimum_phase_order: 511
use_cep_ap: True

[Architecture]
switch_to_tensorflow: False
switch_to_keras: False
hidden_layer_size  : [1024, 1024, 1024, 1024, 1024, 1024]
hidden_layer_type  : ['TANH', 'TANH', 'TANH', 'TANH', 'TANH', 'TANH']

model_file_name: feed_forward_6_tanh

#if RNN or sequential training is used, please set sequential_training to True.
sequential_training : False

dropout_rate : 0.0
batch_size   : 256

# options: -1 for exponential decay, 0 for constant learning rate, 1 for linear decay
lr_decay      : -1 
learning_rate : 0.002

# options: sgd, adam, rprop
optimizer : sgd

warmup_epoch    : 10
training_epochs : 25

[Streams]
# which feature to be used in the output
output_features      : ['mgc', 'lf0', 'vuv', 'bap']
gen_wav_features     : ['mgc', 'lf0', 'bap']

[Data]
train_file_number: 1000
valid_file_number: 66
test_file_number: 66
#buffer size of each block of data to
buffer_size: 200000

[Processes]

# Main processes

AcousticModel: True
GenTestList: True

# sub-processes

NORMLAB  : True
MAKECMP: False
NORMCMP: False

TRAINDNN: False
DNNGEN   : True

GENWAV   : True
CALMCD: False
```

**Comment by Luzeru** [comment]
@martedva  hello:
![image](https://user-images.githubusercontent.com/24689035/46255808-89465980-c4d4-11e8-9d20-40b377910bc7.png)

I have encountered the same problem.
Can you tell me how do you solve it?
